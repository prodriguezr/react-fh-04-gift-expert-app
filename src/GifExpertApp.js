import { useState } from 'react';

import { AddCategory, GifGrid } from './components';

export const GifExpertApp = (defaultCategories = []) => {
  console.log(`${JSON.stringify(defaultCategories)}`);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
