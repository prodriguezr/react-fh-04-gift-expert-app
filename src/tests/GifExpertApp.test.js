import { shallow } from 'enzyme';

import { GifExpertApp } from '../GifExpertApp';

describe('Tests on <GifExpertApp />', () => {
  test('It should display correctly', () => {
    const categories = ['One Punch', 'Dragon Ball Z'];

    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
