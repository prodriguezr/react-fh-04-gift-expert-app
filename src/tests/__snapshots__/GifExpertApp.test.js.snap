// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifExpertApp /> It should display correctly 1`] = `
<Fragment>
  <h2>
    GifExpert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
